src/particles/elements/ChrQuad.H

```diff
@@ -0,0 +1,207 @@
+#define IMPACTX_CHRQUAD_H
+
+#include "particles/ImpactXParticleContainer.H"
+#include "mixin/beamoptic.H"
+#include "mixin/thick.H"
+#include "mixin/nofinalize.H"
+
+#include <AMReX_Extension.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Print.H>      // for PrintToFile
+
+#include <cmath>
+
+
+namespace impactx
+{
+    struct ChrQuad
+    : public elements::BeamOptic<ChrQuad>,
+      public elements::Thick,
+      public elements::NoFinalize
+    {
+        static constexpr auto name = "ChrQuad";
+        using PType = ImpactXParticleContainer::ParticleType;
+
+        /** A Quadrupole magnet with chromatic focusing
+         *
+         * @param ds Segment length in m.
+         * @param k  Quadrupole strength in m^(-2) (MADX convention)
+         *           = (gradient in T/m) / (rigidity in T-m)
+         *        OR Quadrupole strength in T/m (MaryLie convention)
+         *           k > 0 horizontal focusing
+         *           k < 0 horizontal defocusing
+         * @param unit  Unit specification
+         *           unit = 0 MADX convention (default)
+         *           unit = 1 MaryLie convention
+         * @param nslice number of slices used for the application of space charge
+         */
+        ChrQuad( amrex::ParticleReal const ds, amrex::ParticleReal const k,
+              int const unit, int const nslice )
+        : Thick(ds, nslice), m_k(k), m_unit(unit)
+        {
+        }
+
+        /** Push all particles */
+        using BeamOptic::operator();
+
+        /** This is a chrquad functor, so that a variable of this type can be used like a chrquad function.
+         *
+         * @param p Particle AoS data for positions and cpu/id
+         * @param px particle momentum in x
+         * @param py particle momentum in y
+         * @param pt particle momentum in t
+         * @param refpart reference particle
+         */
+        AMREX_GPU_HOST_DEVICE AMREX_FORCE_INLINE
+        void operator() (
+                PType& AMREX_RESTRICT p,
+                amrex::ParticleReal & AMREX_RESTRICT px,
+                amrex::ParticleReal & AMREX_RESTRICT py,
+                amrex::ParticleReal & AMREX_RESTRICT pt,
+                RefPart const & refpart) const {
+
+            using namespace amrex::literals; // for _rt and _prt
+
+            // access AoS data such as positions and cpu/id
+            amrex::ParticleReal const x = p.pos(RealAoS::x);
+            amrex::ParticleReal const y = p.pos(RealAoS::y);
+            amrex::ParticleReal const t = p.pos(RealAoS::t);
+
+            // length of the current slice
+            amrex::ParticleReal const slice_ds = m_ds / nslice();
+
+            // access reference particle values to find beta
+            amrex::ParticleReal const bet = refpart.beta();
+
+            // normalize quad units to MAD-X convention if needed
+            amrex::ParticleReal g = m_k;
+            if (m_unit == 1) {
+                  g = m_k/refpart.rigidity_Tm();
+        };
```

Comment:
Minor formatting and no semicolon:
```suggestion
            if (m_unit == 1) {
                  g = m_k / refpart.rigidity_Tm();
            }
```